File: leetcode/数组与字符串/1.两数之和/leetcode_1.py
from typing import List


class Solution:
    @staticmethod
    def twoSum(nums: List[int], target: int) -> List[int]:
        item_dict = {}
        for k, v in enumerate(nums):
            if target - v in item_dict:
                return [item_dict[target - v], k]
            item_dict[v] = k
        return []


if __name__ == '__main__':
    # res1 = Solution.twoSum([2,7,11,15], 9)
    res1 = Solution.twoSum([3, 2, 4], 6)
    print(res1)

File: leetcode/数组与字符串/1.两数之和/leetcode_1.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	itemDict := map[int]int{}
	for i, v := range nums {
		if p, ok := itemDict[target-v]; ok {
			return []int{p, i}
		}
		itemDict[v] = i
	}
	return nil
}

func main() {
	res1 := twoSum([]int{2, 7, 11, 15}, 9)
	fmt.Println(res1)
}
